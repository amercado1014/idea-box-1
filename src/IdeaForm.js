import React, { Component } from 'react';

class IdeaForm extends Component {
  constructor() {
    super()
    this.state = {
      title: '',
      description: ''
    }
  }

  handleInputChange = (event) => {
    const { value, name } = event.target
    this.setState({[name]: value})
  }

  render() {
    return(
      <div className="IdeaForm">
        <input value={this.state.title} name='title' onChange={this.handleInputChange} placeholder='Title'/>
        <input value={this.state.description} name='description' onChange={this.handleInputChange} placeholder='Description'/>
        <button onClick={() => this.props.addIdea(this.state)}>Submit</button>
      </div>
    )
  }
}

export default IdeaForm;
