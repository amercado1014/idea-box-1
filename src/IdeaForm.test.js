import React from 'react'
import { shallow, mount } from 'enzyme'
import IdeaForm from './IdeaForm'

describe('IdeaForm', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<IdeaForm />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should update the state when handle change is called', () => {
    const wrapper = shallow(<IdeaForm />)
    const mockEvent = { target: { value: 'some string', name: 'title' } }
    wrapper.instance().handleInputChange(mockEvent)
    expect(wrapper.state('title')).toEqual('some string')
  })

  //it('should not update the state if handle change is called with no name', () => {
  //  const wrapper = shallow(<IdeaForm />)
  //  const expected = {...wrapper.state}
  //  const mockEvent = { target: { value: 'val' } }
  //  wrapper.instance().handleInputChange(mockEvent)
  //  expect(wrapper.state()).toEqual(expected)
  //})
  
  it('should call addIdea onHandleSubmit', () => {
    const mockAddIdea = jest.fn()
    const mockEvent = { preventDefault: jest.fn() }
    const wrapper = shallow(<IdeaForm addIdea={mockAddIdea} />)
    wrapper.instance().handleSubmit(mockEvent)
    expect(mockAddIdea).toHaveBeenCalled()
  })

  it('should reset state onHandleSubmit', () => {
  })

})
