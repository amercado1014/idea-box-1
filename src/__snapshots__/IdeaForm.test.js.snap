// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="title"
          name="title"
          onChange={[Function]}
          placeholder="Title"
          value=""
/>,
        <input
          className="desc"
          name="description"
          onChange={[Function]}
          placeholder="Description"
          value=""
/>,
        <button
          className="submitButton"
          type="submit"
>
          Submit
</button>,
      ],
      "className": "IdeaForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "title",
          "name": "title",
          "onChange": [Function],
          "placeholder": "Title",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "desc",
          "name": "description",
          "onChange": [Function],
          "placeholder": "Description",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "submitButton",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="title"
            name="title"
            onChange={[Function]}
            placeholder="Title"
            value=""
/>,
          <input
            className="desc"
            name="description"
            onChange={[Function]}
            placeholder="Description"
            value=""
/>,
          <button
            className="submitButton"
            type="submit"
>
            Submit
</button>,
        ],
        "className": "IdeaForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "title",
            "name": "title",
            "onChange": [Function],
            "placeholder": "Title",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "desc",
            "name": "description",
            "onChange": [Function],
            "placeholder": "Description",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "submitButton",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
